### PR TITLE
[OAP-Cache-OAP][POAE7-305] Add external cache enable config

### PR DESCRIPTION
**Pull request by jikunshang**
## What changes were proposed in this pull request?

add a config to decide whether we need cache data when using external cache

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala [hunk 1]
```diff
@@ -995,12 +996,19 @@ class ExternalCache(fiberType: FiberType) extends OapCache with Logging {
...
-      fiberCache
+      if (cacheEnable) {
+        val fiberCache = cache(fiberId)
+        cacheMissCount.addAndGet(1)
+        fiberSet.add(fiberId)
+        fiberCache.occupy()
+        cacheGuardian.addRemovalFiber(fiberId, fiberCache)
+        fiberCache
+      }
+      else {
```
Comment (yeyuqiang) [inline oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala:1007]: code style, with one line

File: oap-cache/oap/src/main/scala/org/apache/spark/sql/internal/oap/OapConf.scala [hunk 2]
```diff
@@ -433,6 +433,13 @@ object OapConf {
       .intConf
       .createWithDefault(1)
 
+  val OAP_EXTERNAL_CACHE_ENABLE =
+    SqlConfAdapter.buildConf("spark.sql.oap.cache.external.enable")
```
Comment (winningsix) [inline oap-cache/oap/src/main/scala/org/apache/spark/sql/internal/oap/OapConf.scala:437]: Hmm, I would suggest to rename to "spark.sql.oap.cache.external.read.only.enabled" and make it default to false.

File: oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala [hunk 3]
```diff
@@ -995,12 +996,18 @@ class ExternalCache(fiberType: FiberType) extends OapCache with Logging {
...
+      if (cacheEnable) {
+        val fiberCache = cache(fiberId)
+        cacheMissCount.addAndGet(1)
+        fiberSet.add(fiberId)
+        fiberCache.occupy()
+        cacheGuardian.addRemovalFiber(fiberId, fiberCache)
+        fiberCache
+      } else {
+        val fiberCache = super.cache(fiberId)
+        fiberCache.fiberId = fiberId
```
Comment (winningsix) [inline oap-cache/oap/src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala:1008]: Why do we need to do this again? It's set in ```cache``` method already.

**Comment by winningsix** [comment]
@jikunshang comments are not addressed yet.

**Comment by jikunshang** [comment]
> @jikunshang comments are not addressed yet.

Pushed to a wrong repo just now, please delete intel-bigdata/oap/add_external_cache_enable branch.

**Comment by winningsix** [comment]
LGTM +1

**Comment by yeyuqiang** [comment]
LGTM

**Pull request merged**